src/main/java/apod/ApodFrame.java

```diff
@@ -75,41 +94,56 @@ public ApodFrame() {
         photo = new JLabel();
         photoPanel.add(photo);
 
-        prevYear = new JButton("Previous Year");
-        prevYear.addActionListener(this::onSubmitClickPrev);
-        leftPanel.add(prevYear, BorderLayout.WEST);
+        video = new JLabel();
+        photoPanel.add(video);
 
-        nextYear = new JButton("Following Year");
-        nextYear.addActionListener(this::onSubmitClickNext);
-        rightPanel.add(nextYear, BorderLayout.EAST);
+        downloadButton = new JButton("Save Photo");
+        downloadButton.addActionListener(this::onSubmitClicked);
+        topPanel.add(downloadButton);
 
-        AstronomyPicOfDayServiceFactory factory = new AstronomyPicOfDayServiceFactory();
-        presenter = new ApodPresenter(this, factory.getInstance());
-
-    }
 
-    private void onSubmitClickPrev(ActionEvent actionEvent) {
-        LocalDate newDate = datePicker.getDate().minusYears(1);
-        datePicker.setDate(newDate);
-        presenter.loadFromDate(datePicker.getDate().toString());
     }
 
-    private void onSubmitClickNext(ActionEvent actionEvent) {
-        LocalDate newDate = datePicker.getDate().plusYears(1);
-        datePicker.setDate(newDate);
-        presenter.loadFromDate(datePicker.getDate().toString());
-    }
-
-    private void onSubmitClick(ActionEvent actionEvent) {
+    private void onDateChosen(DateChangeEvent dateChangeEvent) {
         presenter.loadFromDate(datePicker.getDate().toString());
     }
 
     public void setPhoto(BufferedImage image) {
+        video.setVisible(false);
+        photo.setVisible(true);
         Image scaledImage = image.getScaledInstance(650, 500, Image.SCALE_SMOOTH);
         ImageIcon imageIcon = new ImageIcon(scaledImage);
         photo.setIcon(imageIcon);
+        downloadButton.setVisible(true);
+    }
+
+    public void setVideoUrl(URL videoUrl) {
+        photo.setVisible(false);
+        downloadButton.setVisible(false);
+        video.setVisible(true);
+        video.setForeground(Color.BLUE);
+        video.setText(videoUrl.toString());
+        video.setCursor(Cursor.getPredefinedCursor(Cursor.HAND_CURSOR));
+
+        video.addMouseListener(new MouseAdapter() {
+            public void mouseClicked(MouseEvent e) {
+                if (e.getClickCount() > 0) {
+                    if (Desktop.isDesktopSupported()) {
+                        Desktop desktop = Desktop.getDesktop();
+                        try {
+                            URI uri = new URI(videoUrl.toString());
+                            desktop.browse(uri);
+                        } catch (IOException | URISyntaxException ex) {
+                            ex.printStackTrace();
+                        }
+                    }
+                }
+            }
+        });
```

Comment:
Doing this adds the MouseListener more than once when `setVideoUrl()` is called again.

Only add the mouse listener once.